applications/ecallmgr/src/ecallmgr_fs_channel.erl

```diff
@@ -597,102 +330,14 @@ fetch_remote(UUID) ->
-    ecallmgr_fs_channels:updates(UUID, props_to_update(Props));
-process_specific_event(<<"CHANNEL_BRIDGE">>, UUID, Props, _) ->
-    OtherLeg = get_other_leg(UUID, Props),
-    ecallmgr_fs_channels:updates(UUID, props:filter_undefined(
-                                         [{#channel.other_leg, OtherLeg}
-                                          | update_callee(UUID, Props)
-                                         ]
-                                        )
-                                ),
-    ecallmgr_fs_channels:update(OtherLeg, #channel.other_leg, UUID);
-process_specific_event(<<"CHANNEL_UNBRIDGE">>, UUID, Props, _) ->
-    OtherLeg = get_other_leg(UUID, Props),
-    ecallmgr_fs_channels:update(UUID, #channel.other_leg, 'undefined'),
-    ecallmgr_fs_channels:update(OtherLeg, #channel.other_leg, 'undefined');
-process_specific_event(_EventName, _UUID, _Props, _Node) ->
-    'ok'.
-
--spec maybe_publish_channel_state(kz_proplist(), atom()) -> 'ok'.
-maybe_publish_channel_state(Props, Node) ->
-    %% NOTE: this will significantly reduce AMQP request however if a ecallmgr
-    %%   becomes disconnected any calls it previsouly controlled will not produce
-    %%   CDRs.  The long-term strategy is to round-robin CDR events from mod_kazoo.
-    Event = ecallmgr_call_events:get_event_name(Props),
-    case props:is_true(<<"Publish-Channel-State">>, Props, 'true')
-        andalso ecallmgr_config:get_boolean(<<"publish_channel_state">>, 'true', Node) of
-        'false' -> lager:debug("not publishing channel state ~s", [Event]);
-        'true' ->
-            case ecallmgr_config:get_boolean(<<"restrict_channel_state_publisher">>, 'false') of
-                'false' -> ecallmgr_call_events:process_channel_event(Props);
-                'true' -> maybe_publish_restricted(Props)
-            end
-    end.
-
--spec maybe_publish_restricted(kz_proplist()) -> 'ok'.
-maybe_publish_restricted(Props) ->
-    EcallmgrNode = kz_term:to_binary(node()),
-    Event = ecallmgr_call_events:get_event_name(Props),
-
-    case props:get_value(?GET_CCV(<<"Ecallmgr-Node">>), Props) of
-        'undefined' -> ecallmgr_call_events:process_channel_event(Props);
-        EcallmgrNode -> ecallmgr_call_events:process_channel_event(Props);
-        _EventEcallmgr ->
-            lager:debug("channel state ~s for call controlled by another ecallmgr(~s), not publishing", [Event, _EventEcallmgr])
-    end.
-
 -spec props_to_record(kz_proplist(), atom()) -> channel().
 props_to_record(Props, Node) ->
     UUID = props:get_value(<<"Unique-ID">>, Props),
     CCVs = ecallmgr_util:custom_channel_vars(Props),
     OtherLeg = get_other_leg(props:get_value(<<"Unique-ID">>, Props), Props),
     #channel{uuid=UUID
             ,destination=props:get_value(<<"Caller-Destination-Number">>, Props)
-            ,direction=kzd_freeswitch:call_direction(Props)
+            ,direction=kz_evt_freeswitch:call_direction(Props)
```

Comment:
These calls to `kz_evt_freeswitch` expect `kz_json:object()`.

Reply:
removing `props_to_record`

Reply:
@lazedo still here, called in `renew/2` as well.